src/geocel/ScopedGeantLogger.cc

```diff
@@ -113,7 +113,7 @@ class GeantLoggerAdapter : public G4coutDestination
   public:
     // Assign to Geant handlers on construction
     GeantLoggerAdapter();
-    ~GeantLoggerAdapter();
+    ~GeantLoggerAdapter() override;
```

Comment:
Why is this override? It's not like you can declare a destructor "final"...

Reply:
To show that the base class has a virtual dtor. This is following the discussion https://github.com/isocpp/CppCoreGuidelines/pull/1448 `virtual` instead of `override` is also fine.

Reply:
That's weird. The parent class doesn't have a `~GeantLoggerAdapter` method. Maybe next c++ version will name the destructor `__del__` so you can actually tell it's overriding... I think I prefer `virtual` here since it says "polymorphic and safe to destruct" rather than "I'm replacing what the parent classdoes".

Reply:
The parent `G4coutDestination` still has a virtual destructor. Even though technically destructor override their parent's I agree that semantically `virtual` makes sense since they are chained and don't replace the parent's behavior.  However, the advantage of `override` is that it will result in an error if the parent dtor is no longer virtual. Using `virtual` would silently redeclare it virtual in the derived which would be confusing.

Anyway, in that specific case, it's not really important since this is a TU-local class but in general I prefer the destructor be declared override if it needs to be defined again in the derived class for the reason above.

Reply:
Declaring as `final` sounds good to me, and now that I know about this behavior I think `override` for virtual destructors is sensible. (It's partly weird because the daughter classes will automatically define virtual `override` destructors...)